src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -51,7 +51,7 @@ public abstract class NLPProcessor extends AbstractProcessor {
 
     private final Environment environment;
 
-    public NLPProcessor(
+    public InferenceProcessor(
```

Comment:
check comments related to this in the main branch PR.

Reply:
Please checkout the comments in 353.